lib/ignored-paths.js

```diff
@@ -201,6 +201,12 @@ class IgnoredPaths {
 
         const ig = ignore().add(DEFAULT_IGNORE_DIRS);
 
+        if (this.options.dotfiles !== true) {
+
+            // Ignore hidden folders.  (This cannot be ".*", or else it's not possible to unignore hidden files)
+            ig.add([".*/*", "!../"]);
```

Comment:
As a separate issue (probably for a different PR) maybe we should name this variable to something other than `ig`. I was having trouble understanding what this code was doing because it's hard to tell what `ig.add` means conceptually.